docs/src/content/hardhat-runner/docs/guides/test-contracts.md

```diff
@@ -1,3 +1,197 @@
-# Testing your contracts
+# Testing contracts
 
-\<New guide, TODO>
+After [compiling your contracts](./compile-contracts.md), the next step is to write some tests to verify that they work as intended. This guide explains our recommended approach for testing contracts in Hardhat. It relies on [ethers](https://docs.ethers.io/v5/) to connect to the Hardhat Network and [Mocha](https://mochajs.org/) and [Chai](https://www.chaijs.com/) to write the tests. It also uses our custom [Chai matchers](/hardhat-chai-matchers) and our [Hardhat Network Helpers](/hardhat-network-helpers) to make it easier to write contracts-related tests.
+
+While this is the way we recommend, Hardhat is flexible and you can customize the approach or take a completely different path with other tools.
+
+### Initial setup
+
+In this guide we’ll write some tests for the sample project so, if you haven’t done it yet, go and [initialize it](./project-setup.md).
+
+We also recommend you use [TypeScript](./typescript.md) to get better autocompletion and catch possible errors earlier. This guide will assume you are using TypeScript, but you can click the tabs of the snippets to switch their language.
+
+The setup includes some example tests in the `test/Lock.ts` file, but ignore them for now. Instead, create a `test/my-tests.ts` file directory and run only those. For example, if your test file is called `test/my-tests.ts`, you can run only that file with `npx hardhat test test/my-tests.ts`
```

Comment:
```suggestion
The setup includes some example tests in the `test/Lock.ts` file, but ignore them for now. Instead, create a `test/my-tests.ts` file. During this guide we'll only run those, by running `npx hardhat test test/my-tests.ts`, instead of just `npx hardhat test`.
```